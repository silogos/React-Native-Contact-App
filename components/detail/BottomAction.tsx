import React from "react";
import { StyleSheet } from "react-native";
import { Link } from "expo-router";
import BottomActionButton from "./BottomActionButton";
import { ThemedView } from "@/components/ThemedView";
import { useThemeColor } from "@/hooks/useThemeColor";

type BottomActionType = {
  contactId: String;
  onDelete: () => void;
  onShare: () => void;
};

export default function BottomAction({
  contactId,
  onDelete,
  onShare,
}: BottomActionType) {
  const color = useThemeColor({}, "border");

  return (
    <ThemedView
      colorName="card"
      style={{ ...styles.container, borderTopColor: color }}
    >
      <Link push href={`edit/${contactId}`} asChild>
        <BottomActionButton iconName="pencil-outline" title="Edit" />
      </Link>
      <BottomActionButton
        onPress={onDelete}
        iconName="trash-bin-outline"
        title="Delete"
      />
      <BottomActionButton
        onPress={onShare}
        iconName="share-outline"
        title="Share"
      />
    </ThemedView>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "space-around",
    height: 64,
    elevation: 4,
    borderTopWidth: 1,
  },
});
