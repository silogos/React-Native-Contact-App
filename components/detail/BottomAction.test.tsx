import React from "react";
import { fireEvent, render } from "@testing-library/react-native";

import BottomAction from "./BottomAction";

describe("Detail", () => {
  it("renders correctly", () => {
    const elm = render(
      <BottomAction contactId={"123"} onDelete={() => {}} onShare={() => {}} />
    ).toJSON();

    expect(elm).toMatchSnapshot();
  });

  it("Render with title", () => {
    const onDelete = jest.fn();
    const onShare = jest.fn();
    const { getByTestId } = render(
      <BottomAction contactId={"123"} onDelete={onDelete} onShare={onShare} />
    );
    const editButton = getByTestId("edit-button");
    const deleteButton = getByTestId("delete-button");
    const shareButton = getByTestId("share-button");

    expect(editButton).toBeTruthy();

    fireEvent(deleteButton, "press");
    expect(onDelete).toHaveBeenCalledTimes(1);

    fireEvent(shareButton, "press");
    fireEvent(shareButton, "press");
    expect(onShare).toHaveBeenCalledTimes(2);
  });
});
