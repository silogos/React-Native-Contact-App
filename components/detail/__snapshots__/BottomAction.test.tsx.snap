// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "borderTopColor": "#e4e4e7",
        "borderTopWidth": 1,
        "elevation": 4,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-around",
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    href="edit/123"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="link"
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "gap": 5,
        "justifyContent": "center",
      }
    }
    testID="edit-button"
  >
    <Text />
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 10,
          },
        ]
      }
    >
      Edit
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "gap": 5,
        "justifyContent": "center",
      }
    }
    testID="delete-button"
  >
    <Text />
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 10,
          },
        ]
      }
    >
      Delete
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "gap": 5,
        "justifyContent": "center",
      }
    }
    testID="share-button"
  >
    <Text />
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 10,
          },
        ]
      }
    >
      Share
    </Text>
  </View>
</View>
`;
